File: libs/as-sdk-integration-tests/assembly/tally.ts
import { JSON } from "json-as";
import { Bytes, OracleProgram, Process, Tally } from "../../as-sdk/assembly";

export class TestTallyVmReveals extends OracleProgram {
	tally(): void {
		const reveals = Tally.getReveals();

		Process.success(Bytes.fromUtf8String(JSON.stringify(reveals)));
	}
}

export class TestTallyVmRevealsFiltered extends OracleProgram {
	tally(): void {
		const reveals = Tally.getConsensusReveals();

		Process.success(Bytes.fromUtf8String(JSON.stringify(reveals)));
	}
}

File: libs/go-sdk-integration-tests/tests/tally.go
package tests

import (
	"encoding/json"
	"fmt"

	"github.com/seda-protocol/seda-sdk/process"
	"github.com/seda-protocol/seda-sdk/tally"
)

func TestTallyVmReveals() {
	reveals, err := tally.GetUnfilteredReveals()
	if err != nil {
		process.Error([]byte(fmt.Sprintf("Error getting reveals: %v", err)))
	}

	encodedReveals, err := json.Marshal(reveals)
	if err != nil {
		process.Error([]byte(fmt.Sprintf("Error marshalling reveals: %v", err)))
	}

	process.Success(encodedReveals)
}

func TestTallyVmRevealsFiltered() {
	reveals, err := tally.GetFilteredReveals()
	if err != nil {
		process.Error([]byte(fmt.Sprintf("Error getting reveals: %v", err)))
	}

	encodedReveals, err := json.Marshal(reveals)
	if err != nil {
		process.Error([]byte(fmt.Sprintf("Error marshalling reveals: %v", err)))
	}

	process.Success(encodedReveals)
}
